providers/entries.rb

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env ruby
+#
+# Author:: Barney Hanlon (<shrikeh@gmail.com>)
+# Resource:: host
+#
+# Copyright 2015, Barney Hanlon
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+require_relative '../libraries/ssh_known_hosts'
+
+# Override Load Current Resource
+def load_current_resource
+  @current_resource = Chef::Resource::SshUserKnownHostsEntries.new(
+    @new_resource.name
+  )
+
+  # A common step is to load the current_resource instance variables with what
+  # is established in the new_resource.
+  # What is passed into new_resouce via our recipes, is not automatically passed
+  # to our current_resource.
+  # DSL converts our parameters/attrbutes to methods to get and set the instance
+  # variable inside the Provider and Resource.
+  @current_resource.entries(@new_resource.entries)
+  @current_resource.path(@new_resource.path)
+  @current_resource.append(@new_resource.append)
+  @current_resource.owner(@new_resource.owner)
+  @current_resource.group(@new_resource.group)
+  @current_resource.mode(@new_resource.mode)
+end
+
+# Helper module for Chef
+module CreateUserKnownHostEntries
+  include SshKnownHosts
+
+  module_function
+
+  @regex_entry = Regexp.new('([^\s]+) ([a-z0-9-]+) ([0-9A-Za-z/+]+[=]+)')
```

Comment:
([0-9A-Za-z/+]+[=]+) needs to be ([0-9A-Za-z/+]+[=]*)

A base64 encoded string uses = for padding when a the decoded string isn't a multiple of 4. If it's a multiple of 4 then there is no padding